### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by buzlan**
1) connect loader to cart (when i add quantity to the cart , or another operation , and take data from backend we will see loader)
Screenshot_1637071395
2) stylized typeOrderScreen 
Screenshot_1637071457
3) create layout reserveTableScreen 
Screenshot_1637071476
4)Add functionality to reserveTableScreen 
a) select date 
Screenshot_1637071523
Screenshot_1637071532
Screenshot_1637071534
b) select table 
Screenshot_1637071561
Screenshot_1637071562
Screenshot_1637071564
Screenshot_1637071567
c) select time
Screenshot_1637071588
Screenshot_1637071591
Screenshot_1637071593
d) when i selected all fields and set checkbox button "Далее" will be enable to click
Screenshot_1637071672
5) for takeawayScreen we have similar logic except 1 screen
Screenshot_1637071698
Screenshot_1637071750
a) select paidType
Screenshot_1637071776
Screenshot_1637071776
Screenshot_1637071785
Screenshot_1637071788
Screenshot_1637071798
Screenshot_1637071800

### CODE REVIEW

File: src/Screens/OrderTakeawayScreen/orderTakeawayScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    let tempDate = new Date(currentDate);
+
+    if (mode === "date") {
+      let fDate =
+        tempDate.getDate() +
+        "/" +
+        (tempDate.getMonth() + 1) +
+        "/" +
+        tempDate.getFullYear();
```
Comment (BakrID) [inline src/Screens/OrderTakeawayScreen/orderTakeawayScreen.tsx:39]: move it to separate file for Date utils

File: src/Screens/PaidType/paidTypeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { TouchableOpacity } from "react-native-gesture-handler";
+
+import { paidTypeScreenStyles } from "../../styles/paidTypeScreenStyle";
+
+export const paidTypeScreen = ({ navigation, route }) => {
+  const [data, setData] = useState([
+    { name: "Онлайн" },
+    { name: "На месте карточкой" },
+    { name: "На месте наличными" },
+  ]);
```
Comment (BakrID) [inline src/Screens/PaidType/paidTypeScreen.tsx:15]: Don't use state for static data

File: src/Screens/ReserveTableScreen/reserveTableScreen.tsx [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
-  const [open, setOpen] = useState(false);
-  const [value, setValue] = useState(null);
-  const [items, setItems] = useState([
-    { label: "двоих", value: "2" },
-    { label: "четверых", value: "4" },
-    { label: "шестерых", value: "6" },
-    { label: "восьмерых", value: "8" },
-    { label: "десятерых", value: "10" },
-  ]);
+  console.log("ROUTE", route.params);
```
Comment (BakrID) [inline src/Screens/ReserveTableScreen/reserveTableScreen.tsx:29]: remove

File: src/Screens/SelectTableScreen/selectTableScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export const selectTableScreen = ({ navigation, route }) => {
+  const [open, setOpen] = useState(false);
+  const [value, setValue] = useState(route.params?.table || "двоих");
+  const [items, setItems] = useState([
+    { label: "двоих", value: "двоих" },
+    { label: "четверых", value: "четверых" },
+    { label: "шестерых", value: "шестерых" },
+    { label: "восьмерых", value: "восьмерых" },
+    { label: "десятерых", value: "десятерых" },
+  ]);
```
Comment (BakrID) [inline src/Screens/SelectTableScreen/selectTableScreen.tsx:17]: If it's const data - don't use state for that

File: src/Screens/SelectTimeScreen/selectTimeScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    "16:00",
+    "16:30",
+    "17:00",
+    "17:30",
+    "18:00",
+    "18:30",
+    "19:00",
+    "19:30",
+    "20:00",
+  ]);
```
Comment (BakrID) [inline src/Screens/SelectTimeScreen/selectTimeScreen.tsx:33]: The same

**Review by BakrID — CHANGES_REQUESTED** [review]
fix comments

File: src/Screens/PaidType/paidTypeScreen.tsx (hunk 2 above)
Comment (buzlan) [reply src/Screens/PaidType/paidTypeScreen.tsx:15]: fixed

File: src/Screens/OrderTakeawayScreen/orderTakeawayScreen.tsx (hunk 1 above)
Comment (buzlan) [reply src/Screens/OrderTakeawayScreen/orderTakeawayScreen.tsx:39]: fixed

File: src/Screens/SelectTimeScreen/selectTimeScreen.tsx (hunk 5 above)
Comment (buzlan) [reply src/Screens/SelectTimeScreen/selectTimeScreen.tsx:33]: fixed

File: src/Screens/SelectTableScreen/selectTableScreen.tsx (hunk 4 above)
Comment (buzlan) [reply src/Screens/SelectTableScreen/selectTableScreen.tsx:17]: image
 i use setItems for dropDownPicker

Comment (BakrID) [reply src/Screens/SelectTableScreen/selectTableScreen.tsx:17]: ok

**Review by BakrID — APPROVED** [review]

**Pull request merged**